### PR TITLE
Added UResult for hostname, hostid and dirname

### PR DESCRIPTION
**Pull request by 353fc443**
Related to #2464

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
Nice! It looks good, but there is one slight problem with `dirname`, which is that the new error handling does not support the usage string yet. I have a PR almost ready to add initial support for that. I'll try to post that PR today!

**Comment by tertsdiepraam** [comment]
Here it is: https://github.com/uutils/coreutils/pull/2473

**Comment by 353fc443** [comment]
@tertsdiepraam Thanks, Waiting for #2473 to be merged.

**Comment by sylvestre** [comment]
Clippy isn't happy :)

```
error: useless use of `format!`
  --> src/uu/dirname/src/dirname.rs:82:41
   |
82 |         return Err(USimpleError::new(1, format!("missing operand")));
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using `.to_string()`: `"missing operand".to_string()`
   |
   = note: `-D clippy::useless-format` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_format

```

**Comment by 353fc443** [comment]
We could use 
`return Err(UUsageError::new(1, "missing operand".to_string()))`
instead of 
`return Err(USimpleError::new(1, format!("missing operand")))`

from #2473

File: src/uu/dirname/src/dirname.rs [hunk 1]
```diff
@@ -77,11 +79,10 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
             print!("{}", separator);
         }
     } else {
-        show_usage_error!("missing operand");
-        return 1;
+        return Err(USimpleError::new(1, format!("missing operand")));
```
Comment (sylvestre) [inline src/uu/dirname/src/dirname.rs:82]:
```suggestion
        return Err(UUsageError::new(1, "missing operand".to_string()))
```

**Comment by sylvestre** [comment]
I guess you saw that it is failing on freebsd:

```
error[E0432]: unresolved import `uucore::error::UUsageError`
  --> src/uu/dirname/src/dirname.rs:13:30
   |
13 | use uucore::error::{UResult, UUsageError};
   |                              ^^^^^^^^^^^ no `UUsageError` in `error`

error: aborting due to previous error

```

**Comment by tertsdiepraam** [comment]
Cirrus CI was not using the latest master branch. A merge fixed the FreeBSD build!

**Comment by sylvestre** [comment]
oups, thanks

**Pull request merged**